pkg/cloudprovider/vsphere/loadbalancer/README.md

```diff
@@ -0,0 +1,120 @@
+# NSX-T Load Balancer Controller
+
+*Kubernetes load balancer support using NSX-T for the `vSphere` cloud controller
+manager.*
+
+This package enriches the cloud provider interface by implementing the load
+balancing API of the cloud controller for an NSX-T environment.
+
+There are two different modes:
+
+- the *managed* mode manages the load balancer service. The NSX-T load balancer
+  service is only created if it is required. The load balancer service will 
+  also be deleted automatically if not needed anymore.
+  This saves resources if no kubernetes service of type `LoadBalancer` is
+  actually used.
+- the *unmanaged* mode is used if the configuration specifies a load balancer
+  service id. Here only the virtual servers are managed.
+
+The basic assumption is that all nodes are bound to a logical tier1 router.
+Here the load balancer service is attached to. Because there may be only
+one such service here, the configuration of the service must be done
+during the creation of the service. Here the selection of the (t-shirt) size is
+required (S, M or L)  
+For every service port a dedicated virtual server is managed which is
+connected to this load balancer service.
+
+## Features
+
+The load balancer controller part of the vsphere cloud controller manager
+is optional. If no `LoadBalancer` or `LoadBalancerClass` section is given
+in the controller configuration loadbalancing support is disabled.
+If enabled the follwing feature are supported:
+
+### Tagging
+
+All generated NSX-T elements are tagged with the app name of the controller,
+the cluster name and the information from the service. Using this tagging it is
+able to handle recovery of lost or dangling elements and garbage collection of
+unused elements previously generated by the controller, even if the kubernetes
+service object is already (accidentally) gone.
+
+### Load Balancer Classes
+
+This load balancer controller supports the usage of multiple load balancer
+classes. Classes are preconfigured in the configuration file of the cloud
+controller manager. Every class may use another `IPPool` configured in NSX-T.
+This supports the creation of load balancers in different visibility realms,
+for example an `*internet facing* or a *private* load balancer. The
+IPPools must be preconfigured in NSX-T. To select a dedicated loadbalancer
+class the kunernetes service object must be annotated with the annotation:
+
+```yaml
+loadbalancer.vsphere.class: <class name>
```

Comment:
This is not consistent with the code, this should be `loadbalancer.vmware.io/class: <class name>`

Reply:
fixed